conjure-java-undertow-runtime/src/test/java/com/palantir/conjure/java/undertow/runtime/AsyncTargetGeneratedService.java

```diff
@@ -0,0 +1,30 @@
+/*
+ * (c) Copyright 2019 Palantir Technologies Inc. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.palantir.conjure.java.undertow.runtime;
+
+import com.google.common.util.concurrent.ListenableFuture;
+import com.palantir.conjure.java.undertow.lib.BinaryResponseBody;
+import com.palantir.tokens.auth.AuthHeader;
+
+// TODO(ckozak): Delete and replace with generated code when the generator is ready
+public interface AsyncTargetGeneratedService {
+    ListenableFuture<String> string(AuthHeader authHeader);
```

Comment:
picking up the question from above: how am I expected to construct these futures? on what thread pool should I run them?

Reply:
In the ideal case, you do not construct them yourself, but use a ListenableFuture from something else (a dialog async client future, [ListeningExecutorService](https://google.github.io/guava/releases/19.0/api/docs/com/google/common/util/concurrent/ListeningExecutorService.html), perhaps with some transformation.

https://github.com/google/guava/wiki/ListenableFutureExplained#creation

The conjure async request processing feature is designed to allow any ListenableFuture to be sufficient regardless of how the service author constructs it. On completion, we dispatch back to the worker pool to serialize the result (success or failure) the same way we would in the blocking case, regardless of how and where the future executed. The only thing that we execute on the thread which completes the future is `exchange.dispatch(args)`, which sends work to the task pool.